core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -52,7 +55,17 @@ public void refresh() {
 
   @Override
   public PartitionSpec spec() {
-    return PartitionSpec.unpartitioned();
+    return spec;
+  }
+
+  @Override
+  public PartitionSpec spec(int specId) {
+    return spec.specId() == specId ? spec : null;
+  }
+
+  @Override
+  public Iterable<PartitionSpec> specs() {
```

Comment:
What about using Map<Integer, PartitionSpec> instead? That would remove the need to have a lookup method and this method. Plus, we need to build a map of specs for the lookup anyway.

Reply:
Makes sense